exonum_launcher/launcher.py

```diff
@@ -123,19 +123,27 @@ def deploy_all(self) -> None:
 
     def wait_for_deploy(self) -> None:
         """Waits for all the deployments to be completed."""
-        pending_deployments = self.launch_state.pending_deployments()
-        for tx_hashes in pending_deployments.values():
-            self._explorer.wait_for_txs(tx_hashes)
-
-        for artifact in pending_deployments:
-            result = self._explorer.wait_for_deploy(artifact)
-            self.launch_state.complete_deploy(artifact, result)
-
-    def start_all(self) -> None:
+        descriptions: Dict[Artifact, str] = dict()
+        check_for_deploy: List[Artifact] = list()
+        for artifact, tx_hashes in self.launch_state.pending_deployments().items():
+            try:
+                self._explorer.wait_for_txs(tx_hashes)
+                descriptions[artifact] = "deployed is ok"
+                check_for_deploy.append(artifact)  # Should be checked for deploy status since no exception occurs.
+            except ExecutionFailError as error:
+                descriptions[artifact] = str(error)
+            except NotCommittedError as error:
+                descriptions[artifact] = str(error)
+        for artifact in self.launch_state.pending_deployments():
+            result = self._explorer.wait_for_deploy(artifact) if artifact in check_for_deploy else ActionResult.Fail
+            self.launch_state.complete_deploy(artifact, result, descriptions[artifact])
+
+    def start_all(self, skipped_artifacts: List[Artifact] = None) -> None:
         """Starts all the service instances from the provided config."""
         config_loaders = [
             self._artifact_plugins.get(instance.artifact, DefaultInstanceSpecLoader())
             for instance in self.config.instances
+            if not skipped_artifacts or instance.artifact not in skipped_artifacts
```

Comment:
IMO, it could make sense to instantiate `skipped_artifacts` at the start of the method:

```python
skipped_artifacts = skipped_artifacts or []
# method body...
```

Reply:
Good catch.